embassy-stm32/src/i2c/mod.rs

```diff
@@ -2,16 +2,13 @@
 
 use crate::interrupt;
 
+// I2C V2 is gated on the `time` feature because timing facilities are necessary
+// to provide timeout functionality in order to prevent some APIs from stalling indefinitely
 #[cfg_attr(i2c_v1, path = "v1.rs")]
-#[cfg_attr(i2c_v2, path = "v2.rs")]
+#[cfg_attr(all(i2c_v2, feature = "time"), path = "v2.rs")]
 mod _version;
 pub use _version::*;
```

Comment:
@andresovela You probably also need to gate the pub use.... I think. Consider that before this, it would always have a 'mod _version' that was valid, but this is no longer the case.

Reply:
Makes sense

Reply:
CI still complains about the same thing

Reply:
So, basically any types, like State is defined within v1 or v2, so without time, it means these are missing. Maybe you'll have an easier time if you add cfg guards within v2.rs on the specific bits that rely on types from embassy-time?

Reply:
Oh I understand what the error is saying now. Cool, thanks :)